Sources/PackageCollections/Storage/PackageCollectionsStorage.swift

```diff
@@ -226,25 +225,87 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
                     }
                 }
 
-                // TODO: consider some diagnostics / warning for invalid data
-                let collections = blobs.compactMap { data -> PackageCollectionsModel.Collection? in
-                    try? self.jsonDecoder.decode(PackageCollectionsModel.Collection.self, from: data)
+                // FIXME: is there a better way?
```

Comment:
👀  this is an interesting one. I put in a workaround of parallelizing the decoding for large enough set, but the out of-the-box performance is JSONDecoder is not impressive

Reply:
👍